encord/orm/dataset.py

```diff
@@ -54,19 +58,93 @@ class DatasetUsers:
     pass
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(frozen=True)
 class DataClientMetadata:
     payload: dict
 
 
+@dataclasses.dataclass(frozen=True)
+class SignedUrl:
+    signed_url: str
+
+
+@dataclasses.dataclass(frozen=True)
+class ImageData:
+    id: int
+    title: str
+    user_hash: UUID
+    file_link: str
+    file_type: str
+    file_size: float
+    image_hash: UUID
+    storage_location: StorageLocation
+    created_at: datetime
+    last_edited_at: datetime
+    height: int
+    width: int
+
+    @classmethod
+    def from_dict(cls, json_dict: Dict):
+        return ImageData(
+            json_dict["id"],
+            json_dict["user_hash"],
+            json_dict["title"],
+            json_dict["file_link"],
+            json_dict["file_type"],
+            json_dict["file_size"],
+            json_dict["image_hash"],
+            json_dict["storage_location"],
+            parser.parse(json_dict["created_at"]),
+            parser.parse(json_dict["last_edited_at"]),
+            json_dict["height"],
+            json_dict["width"],
+        )
+
+
+@dataclasses.dataclass(frozen=True)
+class ImagesInGroup:
+    images: List[Dict]
+
+
+@dataclasses.dataclass(frozen=True)
+class DicomFileLinks:
+    file_links: List[str]
+
+
+def is_signed_url(string: str) -> bool:
+    pattern = re.compile(r"^https://.*")
+    return pattern.match(string) is not None
+
+
+def check_if_images_have_signed_url(images: List[ImageData]) -> bool:
+    if len(images) == 0:
+        return False
+    return is_signed_url(images[0].file_link)
+
+
+def check_if_file_links_are_signed_urls(file_links: List[str]) -> bool:
+    if len(file_links) == 0:
+        return False
+    return is_signed_url(file_links[0])
+
+
 class DataRow(dict, Formatter):
     def __init__(
         self,
         uid: str,
         title: str,
         data_type: DataType,
         created_at: datetime,
+        last_edited_at: datetime,
+        width: int,
+        height: int,
+        file_link: str,  # file link from db
```

Comment:
I think we should remove or rewrite this comment to be for users of the SDK. This feel now like a comment for us, the developers